pkg/controller/syncer/keycloak.go

```diff
@@ -182,14 +183,20 @@ func (k *KeycloakSyncer) Sync() ([]userv1.Group, error) {
 
 	for _, cachedGroup := range k.CachedGroups {
 
+		groupAttributes := map[string]string{}
+
+		for key, value := range cachedGroup.Attributes {
+			groupAttributes[key] = strings.Join(value, "'")
```

Comment:
1. Validation or formatting should occur as the annotation key must abide by these rules otherwise an error would occur:

https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set

2. It there a reason for `strings.join` here?

3. Attributes are not sent by default on groups. You need to add a parameter to the query to request for all fields of a group `Full: true`

Reply:
1. added standard annotation key validation
2. metadata for keycloak groups is an array of strings and I have to turn it into a string. to be useful one will have to put just one value on the keycloak side.
3. added that parameter.